ontopy/ontology.py

```diff
@@ -409,26 +416,38 @@ def get_by_label_all(self, label, label_annotations=None, prefix=None):
                 f"Invalid label definition, {label!r} contains spaces."
             )
 
-        if label_annotations is None:
-            annotations = (_.name for _ in self.label_annotations)
-        else:
+        if label_annotations:
             annotations = (
                 _.name if hasattr(_, "storid") else _ for _ in label_annotations
             )
-        entity = self.world.search(**{next(annotations): label})
-        for key in annotations:
-            entity.extend(self.world.search(**{key: label}))
+        elif self._label_annotations:
+            annotations = (_.name for _ in self.label_annotations)
```

Comment:
In general it is bad practice to use `_` as a variable name. It is fine to use as a placeholder for an unused variable, but not for something we are using. Reasons: (1) it is commonly used indicate an unused variable, let not give it more semantics and (2) it is assigned to the last result in many interpreters, which can lead to surprises for people that test code snippets by copy-paste into an interpreter. 
```suggestion
            annotations = (la.name for la in self.label_annotations)
```